apps/nextjs/pages/_app.tsx

```diff
@@ -1,8 +1,57 @@
 import '../styles/globals.css';
 import type {AppProps} from 'next/app';
-import {ShopifyProvider, CartProvider} from '@shopify/hydrogen-react';
+import {
+  ShopifyProvider,
+  CartProvider,
+  ShopifyCookies,
+  sendShopifyAnalytics,
+  getClientBrowserParameters,
+  AnalyticsEventName,
+  type ShopifyPageViewPayload,
+} from '@shopify/hydrogen-react';
+import {useRouter} from 'next/router';
+import {useEffect} from 'react';
+
+const analyticsShopData = {
+  shopId: 55145660472,
```

Comment:
No. These analytics components should not care "how" you get the data. It just needs to care about what you supply as inputs to it.

In the case of `shopId`, `currency`, `acceptedLanguage`, I am leaving the decision of how to get this data to developer: It can come from

* Querying from storefront api
* Pulling from `shopify.config.js` .. or
* A constant variable because developer knows this value would never change

I don't want these analytics components to have any dependency on other components like `<ShopifyProvider>` because you can send analytics to Shopify with or without Storefront api.